File: back/back/Models/HistoriqueCampagne.cs
using LiteDB;

namespace back.Models;

public sealed class HistoriqueCampagne
{
     [BsonId]
     public int Id { get; set; }

     [BsonRef]
     public Campagne Campagne { get; set; } = null!;
     
     public string Titre { get; set; }  = null!;
     public string Date { get; set; } = null!;
     public string Texte { get; set; } = null!;
    public string CodeOperation { get; set; } = null!;

    [BsonRef]
    public PlaneteOrigine? Planete { get; set; }

     public List<string> ListeNomFichier { get; set; } = [];
}

File: front/src/models/HistoriqueCampagne.ts
import { PlaneteOrigineLeger } from "./PlaneteOrigine"

export type HistoriqueCampagne =
{
    id: number,
    date: string,
    titre: string,
    texte: string,
    codeOperation: string,
    planete: PlaneteOrigineLeger | null,
    listeUrlImage: string[]
}

export type HistoriqueCampagneRequete =
{
    date: string,
    titre: string,
    codeOperation: string,
    texte: string,
    idPlanete: number
}